client/verta/verta/dataset/_dataset.py

```diff
@@ -93,24 +93,107 @@ def _set_commit_and_blob_path(self, commit, blob_path):
         self._commit = commit
         self._blob_path = blob_path
 
-    def download(self, component_path, download_to_path=None, chunk_size=32*(10**3)):
+    def _get_components_to_download(self, component_path, download_to_path=None):
+        """
+        Identify components to be downloaded, along with their local destination paths.
+
+        Parameters
+        ----------
+        component_path : str
+            Path to directory or file within blob.
+        download_to_path : str, optional
+            Local path to download to.
+
+        Returns
+        -------
+        components_to_download : dict
+            Map of component paths to local destination paths.
+        downloaded_to_path : str
+            Absolute path where file(s) were downloaded to. Matches `download_to_path` if it was
+            provided as an argument.
+
+        """
+        implicit_download_to_path = download_to_path is None
+
+        # look for an exact match with `component_path` as a file
+        for component_blob in self._path_component_blobs:
+            if component_blob.path == component_path:
+                if implicit_download_to_path:
+                    # default to filename from `component_path`, in cwd
+                    local_path = os.path.basename(component_path)
+
+                    # avoid collision with existing file
+                    while os.path.exists(local_path):
+                        local_path = _file_utils.increment_path(local_path)
+                else:
+                    # exactly where the user requests
+                    local_path = download_to_path
+
+                return ({component_blob.path: local_path}, os.path.abspath(local_path))
+
+        # figure out where files are going to be downloaded to
+        if implicit_download_to_path:
+            # automatically determine directory
+            # TODO: this is going have issues with `component_path` in {".", "..", "/"}
```

Comment:
The problem here is if the blob contains
```python
"../a.txt"
"../b.txt"
"../c.txt"
```
then this logic as-is would write to
```python
".. 1/a.txt"
".. 1/b.txt"
".. 1/c.txt"
```

In these cases, I'm going to need to go through the blobs and check their outermost path components against path collisions, and be able to write to
```python
"a.txt"
"b 1.txt"  # b.txt already exists
"c.txt"
```